routes/favicon.coffee

```diff
@@ -0,0 +1,28 @@
+request = require 'request'
+fs = require 'fs'
+cachefile = 'favicon.ico'
+
+favicon = (req, res, next) ->
+	try
+		res.send fs.readFileSync cachefile
+	catch error
+		request "https://vine.co/", (error, response, body) ->
+			return next error if error?
+			unless response.statusCode is 200
+				error = new Error "Unexpected response code #{response.statusCode} while getting vine.co homepage"
+				error.status = 500
+				return next error
+			request (/https:\/\/[^"]*?favicon\.ico/.exec body)[0], {encoding: 'binary'}, (error, response, body) ->
+				return next error if error?
+				unless response.statusCode is 200
+					error = new Error "Unexpected response code #{response.statusCode} while getting favicon"
+					error.status = 500
+					return next error
+				try
+					fs.writeFileSync cachefile, body, "binary"
+				catch error
+					error = new Error "Could not write favicon"
+					return next error
+				favicon req, res, next
```

Comment:
This could indeed have happened when the file was there but not readable. But I eliminated the recursive call while rewriting to the async function variants.